api/base/serializers.py

```diff
@@ -88,7 +89,34 @@ def to_internal_value(self, data):
         return super(JSONAPIListField, self).to_internal_value(data)
 
 
-class JSONAPIHyperlinkedIdentityField(ser.HyperlinkedIdentityField):
+class AuthorizedCharField(ser.CharField):
+    """
+    Passes auth of the logged-in user to the object's method
+    defined as the field source.
+
+    Example:
+        content = AuthorizedCharField(source='return_content')
+    """
+    def __init__(self, source=None, **kwargs):
+        assert source is not None, 'The `source` argument is required.'
+        self.source = source
+        super(AuthorizedCharField, self).__init__(source=self.source, **kwargs)
+
+    def get_attribute(self, obj):
+        user = self.context['request'].user
+        auth = auth_core.Auth(user)
+        field_source_method = getattr(obj, self.source)
+        return field_source_method(auth=auth)
+
+
+class HyperlinkedFieldMixin(object):
```

Comment:
This mixin seems unnecessary; the method that it adds doesn't use `self` at all. Better to avoid multiple inheritance if possible. Why not just make a function, `format_links_object`?